### PR TITLE
Fixes and improvements for shaders

### PR DESCRIPTION
**Pull request by themuflon**

### CODE REVIEW

File: loom/graphics/gfxQuadRenderer.cpp [hunk 1]
```diff
@@ -106,61 +106,15 @@ void QuadRenderer::submit()
...
-                        ctx->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-                        break;
-                    default:
-                        lmAssert(false, "Unsupported smoothing: %d", tinfo.smoothing);
-                }
-                
                 sTextureStateValid = true;
             }
 
+            // This is not really perdictable and does not mess up the graphics
```
Comment (SmilyOrg) [inline loom/graphics/gfxQuadRenderer.cpp]: perdictable -> predictable

Also I don't understand what this comment means?

Comment (themuflon) [inline loom/graphics/gfxQuadRenderer.cpp]: We store some  graphics state stuff so we don't rebind stuff that's already bound. This only applies to GL_TEXTURE0 though.

Comment (SmilyOrg) [inline loom/graphics/gfxQuadRenderer.cpp]: Ah, can you add something along those lines to the comment so it's clearer? Thanks :+1:

**Comment by SmilyOrg** [comment]
:+1:

**Pull request merged**